lib/Controller/EndpointController.php

```diff
@@ -96,35 +97,37 @@ public function listNotifications() {
 			$data[] = $this->notificationToArray($notificationId, $notification);
 		}
 
-		return new \OC_OCS_Result(
-			$data,
-			100, // HTTP::STATUS_OK, TODO: <status>failure</status><statuscode>200</statuscode>
-			null,
-			['ETag' => $this->generateEtag($notificationIds)]
-		);
+		$etag = $this->generateEtag($notificationIds);
+		/**
+		 * Not documented yet, check with clients
+		if ($this->request->getHeader('ETag') === $etag) {
+			return new DataResponse($data, Http::STATUS_NOT_MODIFIED);
+		}
+		 */
```

Comment:
Well anyway, I'm not sure the client sends the etag, nor whether they handle 304 correctly, because the docs didn't say anything about 304 yet